json-schema-processor/src/main/java/io/micronaut/jsonschema/visitor/JsonSchemaVisitor.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.jsonschema.visitor;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.micronaut.core.annotation.AnnotationValue;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.inject.ast.ClassElement;
+import io.micronaut.inject.ast.EnumElement;
+import io.micronaut.inject.ast.PropertyElement;
+import io.micronaut.inject.ast.TypedElement;
+import io.micronaut.inject.visitor.TypeElementVisitor;
+import io.micronaut.inject.visitor.VisitorContext;
+import io.micronaut.inject.writer.GeneratedFile;
+import io.micronaut.jsonschema.JsonSchema;
+import io.micronaut.jsonschema.visitor.JsonSchemaConfigurationVisitor.JsonSchemaContext;
+import io.micronaut.jsonschema.visitor.aggregator.DocumentationInfoAggregator;
+import io.micronaut.jsonschema.visitor.aggregator.JacksonInfoAggregator;
+import io.micronaut.jsonschema.visitor.aggregator.SchemaInfoAggregator;
+import io.micronaut.jsonschema.visitor.aggregator.ValidationInfoAggregator;
+import io.micronaut.jsonschema.visitor.model.Schema;
+import io.micronaut.jsonschema.visitor.model.Schema.Type;
+import io.micronaut.jsonschema.visitor.serialization.JsonSchemaMapperFactory;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.net.URI;
+import java.util.*;
+
+import static io.micronaut.jsonschema.visitor.JsonSchemaConfigurationVisitor.JSON_SCHEMA_CONFIGURATION_PROPERTY;
+
+/**
+ * A visitor for creating JSON schemas for beans.
+ * The bean must have a {@link JsonSchema} annotation.
+ *
+ * @since 1.0.0
+ * @author Andriy Dmytruk
+ */
+@Internal
+public final class JsonSchemaVisitor implements TypeElementVisitor<JsonSchema, Object> {
```

Comment:
I don't think this processor should be aggregating. It should be `ISOLATING` since each schema file originates from a single class. The `JsonSchemaConfigurationVisitor` should be deleted and the configuration values it provides should be read from annotation processor arguments (override `getSupportedOptions()` in this class) which provide annotation processor inputs.

Reply:
Still, it seems like it is `AGGREGATING`, as if do something like `beanProperty.getGenericType().getAnnotation(JsonSchema.class)` this might change in the property type. If the json schema changes of the property type changes its id, we need to change the reference.

Reply:
if it is on the property then the originating type is the type that declares the property and there is still only 1